dandischema/models.py

```diff
@@ -58,8 +58,19 @@
 )
 ASSET_UUID_PATTERN = r"^dandiasset:" + UUID_PATTERN
 VERSION_PATTERN = r"\d{6}/\d+\.\d+\.\d+"
-DANDI_DOI_PATTERN = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
-DANDI_PUBID_PATTERN = rf"^DANDI:{VERSION_PATTERN}"
+# Vendored
+# DANDI_DOI_PATTERN = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
+# Unvendored, only with 10. prefix, as likely all would have for datacite
+DANDI_DOI_PATTERN = rf"^10.\d+/[a-z]\.{VERSION_PATTERN}"
+# Vendored:
+# DANDI_PUBID_PATTERN = rf"^DANDI:{VERSION_PATTERN}"
+# Unvendored:
+DANDI_PUBID_PATTERN = rf"^[A-Z]+:{VERSION_PATTERN}"
+# Vendored:
+#  potentially could be set to alternative namespace
+# Unvendored: use "dandi" by default
+DANDI_NSKEY = "dandi"
```

Comment:
I don't think I get this one completely. How are you going to vendorize it if this is hardcoded? Is the user supposed to alter this line for different instances of dandi?

Reply:
I was thinking about just ad-hoc env variables handling here, but @candleindark reminded about https://docs.pydantic.dev/latest/concepts/pydantic_settings/ which  he also used in datalad-registry, so may be we should go for it here... 
benefits:
- validation
- flexibility in import (could be entire config in json)

then, to minimize changes we would just assign constants here.  It is a separate package https://pypi.org/project/pydantic-settings/